server/server.go

```diff
@@ -221,6 +255,7 @@ func (s *Server) Start(ctx context.Context) error {
 	defer cancel()
 	err = server.Shutdown(shutdownCtx)
 	grpcServer.GracefulStop()
+	configGrpcServer.GracefulStop()
 	return errors.Join(err, serverGroup.Wait())
```

Comment:
Instead of joining these errors here why not just log them when they happen? i.e.:

```go
if err := server.Shutdown(shutdownCtx); err != nil {
    logger.Sugar().Errorf("failed to shutdown server: %s", err)
}
if err := serverGroup.Wait(); err != nil {
    logger.Sugar().Errorf("error when waiting to shutdown server: %s", err)
}
return ctx.Err()
```

The the only thing the caller does is take the errors joined here, wrap them again with more text and then print them in a single log statement to the console. If `ctx.Err()` is used the caller can still check that the function was stopped because of the context being cancelled (rather than an actual error that should be handled or logged) and then continue to exit. Wdyt?